src/opnsense/mvc/app/library/OPNsense/Backup/Nextcloud.php

```diff
@@ -0,0 +1,291 @@
+                "help" => gettext("The name you use for logging into your Nextcloud account"),
+                "value" => null
+            ),
+            array(
+                "name" => "nextcloud_password",
+                "type" => "password",
+                "label" => gettext("Password"),
+                "help" => gettext("The app password which has been generated for you"),
+                "value" => null
+            ),
+            array(
+                "name" => "nextcloud_backupdir",
+                "type" => "text",
+                "label" => gettext("Directory Name"),
+                "value" => 'OPNsense-Backup'
+            )
+        );
+        $cnf = Config::getInstance();
+        if ($cnf->isValid()) {
+            $config = $cnf->object();
+            foreach ($fields as &$field) {
+                $fieldname = $field['name'];
+                if (isset($config->system->remotebackup->$fieldname)) {
+                    $field['value'] = (string)$config->system->remotebackup->$fieldname;
+                }
+            }
+        }
+
+        return $fields;
+    }
+
+    /**
+     * backup provider name
+     * @return string user friendly name
+     */
+    public function getName()
+    {
+        return gettext("Nextcloud");
+    }
+
+    /**
+     * validate and set configuration
+     * @param array $conf configuration array
+     * @return array of validation errors when not saved
+     */
+    public function setConfiguration($conf)
+    {
+        $input_errors = array();
+        // format of backup directory name
+        if (isset($_POST['nextcloud_backupdir']) && preg_match('/^[a-z0.9-]+$/i', $_POST['nextcloud_backupdir'])) {
+            $input_errors[] = gettext('The Backup Directory can only consist of alphanumeric characters.');
+        }
+        // required fields
+        if (isset($config['system']['remotebackup']['nextcloud_enabled']) && $config['system']['remotebackup']['nextcloud_enabled'])
+        {
+            $fieldval = array(
+                'nextcloud_url' => gettext('An URL for the Nextcloud server must be set.'),
+                'nextcloud_user' => gettext('An user for the Nextcloud server must be set.'),
+                'nextcloud_password' => gettext('An password for a Nextcloud server must be set.')
+            );
+            foreach ($fieldval as $fieldname => $errormsg) {
+                if (empty($config['system']['remotebackup'][$fieldname])) {
+                    $input_errors[] = $errormsg;
+                }
+            }
+        }
+        if (count($input_errors) == 0) {
+            $config = Config::getInstance()->object();
+            if (!isset($config->system->remotebackup)) {
+                $config->system->remotebackup = array();
+            }
+            foreach ($this->getConfigurationFields() as $field) {
+                $fieldname = $field['name'];
+                if ($field['type'] == 'file') {
```

Comment:
In case you're not using a model (which would need a refactoring of this section), you could ditch the file handling here, the settings don't seem to require file type inputs.